### ISSUE TITLE
[Bug] Skywalking-client-js breaks native fetch implementation when using Request object with headers

### ISSUE THREAD

**Issue by weareoutman** [apache/skywalking#12367]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

NodeJS Client Side Agent (apache/skywalking-client-js)

### What happened

Relevant code: https://github.com/apache/skywalking-client-js/blob/e36b33bb415b7cef6a280d0e4579f9f3327b7d0b/src/trace/interceptors/fetch.ts#L82-L91

<img width="368" alt="image" src="https://github.com/apache/skywalking/assets/2338946/9e6b4596-c107-48f7-a019-c0a828e48a5b">

If we call fetch with a Request object which contains headers, these headers will be dropped if enable skywalking-client-js.

```js
const req = new Request("/api/projects", {
  headers: {
    "my-custom-header": "test",
  },
});

fetch(req);
```

### What you expected to happen

Keep user defined headers.

### How to reproduce

https://codesandbox.io/p/sandbox/skywalking-headers-bug-repro-ss74fd

Open the devtools, switch to network tab, filter Fetch/XHR, open details for the request `/api/projects`, in Request Headers, user defined header `My-Custom-Header` is missing.

While change the value of `enableClientMonitor` to `false` in `monitor.mjs`, the header `My-Custom-Header` exists as expected.

### Are you willing to submit a pull request to fix on your own?

- [ ] Yes I am willing to submit a pull request on my own!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by weareoutman** [comment]
The current implementation also has another problem when using fetch with native `Headers` object.

The following code will result in no `sw8` header sent:

```js
fetch(url, {
  headers: new Headers({ ... })
})
```

**Comment by Fine0830** [comment]
I didn't reproduce this issue, and it works on my side.

![1](https://github.com/apache/skywalking/assets/20871783/80f7dd76-e9ac-492f-b39f-316d2a41eee6)

**Comment by weareoutman** [comment]
According to [the spec](https://fetch.spec.whatwg.org/#request):

<img width="826" alt="image" src="https://github.com/apache/skywalking/assets/2338946/0665185d-990a-4c9d-bf8f-d96d50f61384">

`headers` in `init` (the second parameter for `fetch`) will **override** the headers from `resource` (the first parameter for `fetch`), instead of **extending** it.

**Comment by WHChen-Alex** [comment]
👏🏻

**Comment by weareoutman** [comment]
I'm afraid this has not been fixed by that PR apache/skywalking-client-js#127

It only fixes this:

> The current implementation also has another problem when using fetch with native `Headers` object.
> 
> The following code will result in no `sw8` header sent:
> 
> ```js
> fetch(url, {
>   headers: new Headers({ ... })
> })
> ```

However, passing a Request object that contains headers will result in these headers been dropped as well. @wu-sheng
